src/lib/db.js

```diff
@@ -118,4 +130,29 @@ export class Database {
   connectActivityToCreator(url, creator) {
     return this.db.activity.update(url, { creator: creator });
   }
+
+  logDonation(donation) {
+    return this.db.donations.add(donation);
+  }
+
+  getDonations() {
+    let donations;
```

Comment:
I would like to avoid this kind of state hacking whenever possible but doing it cleanly in this function might be tricky... one way to do it would be:
``` javascript
  getDonations() {
    return this.db.donations
      .reverse()
      .limit(100)
      .toArray()
      .then(ds =>
        Promise.all(
          ds.map(d =>
            this.getCreator(d.url).then(creator =>
              Object.assign(d, { creator: creator.name })
            )
          )
        )
      )
      .catch(err => {
        console.log("Couldn't get donation history from db:", err);
      });
  }
```
but it creates a sort  of "Promise-pyramid", I'd still prefer it over the current solution but will try to think of something better.